code/src/OSFramework/Grid/AbstractDataSource.ts

```diff
@@ -227,11 +227,27 @@ namespace OSFramework.Grid {
         }
 
         public getRowNumberByKey(key: string): number {
-            return this.parentGrid.provider.rows.findIndex(
+            const row = this.parentGrid.provider.rows.findIndex(
                 (item) =>
-                    _.get(item.dataItem, this.parentGrid.config.keyBinding) ===
-                    key
+                    _.get(
+                        item.dataItem,
+                        this.parentGrid.config.keyBinding
+                    ).toString() === key
             );
+
+            // Validation of row to prevent the default row key
+            if (row === -1) {
+                throw new Error(Enum.ErrorMessages.Row_InvalidRowDataKey);
+            }
+
+            // Throws the error when is invalid
+            try {
+                row;
+            } catch (error) {
```

Comment:
This doesn't seem right and is triggering an error on sonarcloud. You should have all the code starting on const row here

Reply:
Already fixed